### ISSUE TITLE
`reportGeneralTypeIssues` false positive when implementing protocol with `Final` attribute

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1205]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoBiqAhgDYA0UACuDGAMZikCwAUG2-acQM4%2BFgwAChpg6jUgEoAXGyjyoxaYRKkA2qhgBdDu1Zde-AELEQQgoJlyFSzCnwBeKAEYgA)

```python
from typing import Final, Protocol


class Foo(Protocol):
    a: Final[int]


class Bar(Foo):
    a: int = 1 # "a" cannot be redeclared because parent class "Foo" declares it as Final  (reportGeneralTypeIssues)
```

this error doesn't make sense because it makes the protocol impossible to implement explicitly (which is different to extending, see https://github.com/DetachHead/basedpyright/issues/294#issuecomment-2611489040)

`Final` sucks because it means both immutable and not overridable for some stupid reason. but in the context of a protocol it should not mean it can't be overridden (aka. implemented) because that wouldn't make sense for a protocol

related: #1061

note that i am ignoring the concept of a default implementation in a protocol because that's the stupidest thing ever and a complete misuse of protocols (see #630)
